zipline/utils/calendars/trading_calendar.py

```diff
@@ -690,11 +699,19 @@ def minute_to_session_label(self, dt, direction="next"):
         pd.Timestamp (midnight UTC)
             The label of the containing session.
         """
+        if direction == "next":
```

Comment:
We no longer have `cache_tools` in zipline, replaced it with `lru-dict`, which is faster.

I don't think it can be used as a decorator out of the box, but that could definitely be added.